### PR TITLE
Fix missing query params for GET/DELETE requests, don't suppress errors

### PR DESCRIPTION
**Pull request by hownowstephen**
The client was suppressing an error when calling `addQueryParams` due to how the variadic params were being passed along to the `get` func. As a result, **GET requests were not actually sending along their associated query parameters**, so for a request to `RatesForLocation` only the zip code was actually being passed along, even if more address information was provided.

Fixes this bug in GET requests and changes the private `get` func to not use variadic parameters as it's not exposed as the public interface, and shouldn't need any extra syntactic sugar around including / not including parameters. Also bubbles up the error from `addQueryParams` to hopefully help prevent regression.

I'd recommend re-evaluating the other places in the library where errors are being suppressed, to ensure there aren't any other opportunities for gotchas like this one

### CODE REVIEW

**Comment by ScottRudiger** [comment]
@hownowstephen Thanks for pointing this out! What you updated so far LGTM, but I'll go ahead and add a commit for the private `delete` func as well since it's in a similar state as `get`.

File: ShowOrder.go [hunk 1]
```diff
@@ -18,25 +18,31 @@ type ShowOrderResponse struct {
...
 type OrderLineItem struct {
-	ID                string  `json:"id,omitempty"`
-	Quantity          int     `json:"quantity,omitempty"`
-	ProductIdentifier string  `json:"product_identifier,omitempty"`
-	Description       string  `json:"description,omitempty"`
-	ProductTaxCode    string  `json:"product_tax_code,omitempty"`
-	UnitPrice         float64 `json:"unit_price,omitempty,string"`
-	Discount          float64 `json:"discount,omitempty,string"`
-	SalesTax          float64 `json:"sales_tax,omitempty,string"`
+	ID                json.Number `json:"id,omitempty"`
```
Comment (ScottRudiger) [inline ShowOrder.go]: The API accepts a string for line item `id`; however, the response is actually a number.  After bubbling up errors from `json.Unmarshal` this issue became visible. With the type changed from `string` to `json.Number`, `ID` may still be passed in as a string, but will not error when converting the response.

**Comment by ScottRudiger** [comment]
@codymullins @kkolk Thanks for helping out with reviewing since I added some commits here as well.

Previously, GET/DELETE requests were not correctly passing query params to the API, so that's why we have a slightly different function signature and related code changes for these methods.

Additionally, we're surfacing more errors that could potentially have been swallowed to avoid similar issues going forward.

Let me know if you have any questions. 👍

File: ShowCustomer.go [hunk 2]
```diff
@@ -13,11 +13,13 @@ type ShowCustomerResponse struct {
 //
 // See https://developers.taxjar.com/api/reference/?go#get-show-a-customer for more details․
 func (client *Config) ShowCustomer(customerID string) (*ShowCustomerResponse, error) {
-	res, err := client.get("customers/" + customerID)
+	res, err := client.get("customers/"+customerID, nil)
```
Comment (codymullins) [inline ShowCustomer.go]: We could use `fmt.Sprintf(...)` in these places (just an FYI).

File: RatesForLocation.go [hunk 3]
```diff
@@ -47,11 +47,17 @@ type RatesForLocationResponse struct {
 //
 // See https://developers.taxjar.com/api/reference/?go#get-show-tax-rates-for-a-location for more details.
 func (client *Config) RatesForLocation(zip string, params ...RatesForLocationParams) (*RatesForLocationResponse, error) {
-	res, err := client.get("rates/"+zip, params)
+	var p interface{}
+	if len(params) > 0 {
+		p = params[0]
+	}
```
Comment (codymullins) [inline RatesForLocation.go]: I'd personally prefer having newlines after constructs like `if`.

**Pull request merged**